unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs

```diff
@@ -225,125 +225,142 @@ public bool ProcessMessage(MessagingBus.QueuedSceneMessage_Scene msgObject, out
-                                scene.RemoveEntity(payload.entityId);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.RemoveEntity payload)
+                            scene.RemoveEntity(payload.entityId);
+                        break;
+                    }
 
                     case MessagingTypes.INIT_DONE:
-                        {
-                            scene.sceneLifecycleHandler.SetInitMessagesDone();
-                            break;
-                        }
+                    {
+                        scene.sceneLifecycleHandler.SetInitMessagesDone();
+                        break;
+                    }
 
                     case MessagingTypes.QUERY:
-                        {
-                            if (msgObject.payload is QueryMessage queryMessage)
-                                ParseQuery(queryMessage.payload, scene.sceneData.id);
-                            break;
-                        }
+                    {
+                        if (msgPayload is QueryMessage queryMessage)
+                            ParseQuery(queryMessage.payload, scene.sceneData.id);
+                        break;
+                    }
 
                     case MessagingTypes.OPEN_EXTERNAL_URL:
-                        {
-                            if (msgObject.payload is Protocol.OpenExternalUrl payload)
-                                OnOpenExternalUrlRequest?.Invoke(scene, payload.url);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.OpenExternalUrl payload)
+                            OnOpenExternalUrlRequest?.Invoke(scene, payload.url);
+                        break;
+                    }
 
                     case MessagingTypes.OPEN_NFT_DIALOG:
-                        {
-                            if (msgObject.payload is Protocol.OpenNftDialog payload)
-                                OnOpenNFTDialogRequest?.Invoke(payload.contactAddress, payload.tokenId, payload.comment);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.OpenNftDialog payload)
+                            OnOpenNFTDialogRequest?.Invoke(payload.contactAddress, payload.tokenId,
+                                payload.comment);
+                        break;
+                    }
 
                     default:
-                        Debug.LogError($"Unknown method {method}");
-                        return true;
+                        throw new Exception($"Unknown method.");
                 }
-
-                ProfilingEvents.OnMessageProcessEnds?.Invoke(method);
-
-#if UNITY_EDITOR
-                OnMessageProcessInfoEnds?.Invoke(sceneId, method);
-#endif
-
-                res = true;
             }
-
-            else
+            catch (Exception e)
```

Comment:
Do this preserves the stack trace?

Reply:
Looks like it does. I just tested it in unity editor and stacktrace was ok.
In browser is also more understandable as well.

With the try/catch:
<img width="573" alt="Screen Shot 2021-01-03 at 18 15 52" src="https://user-images.githubusercontent.com/4195708/103489331-9c061200-4df2-11eb-92d9-de1dd5e3910d.png">

Without it:
<img width="572" alt="Screen Shot 2021-01-03 at 18 26 35" src="https://user-images.githubusercontent.com/4195708/103489343-ac1df180-4df2-11eb-8c82-90cd2d7c5bb5.png">